src/main/java/marquez/db/DatasourceDao.java

```diff
@@ -28,13 +28,16 @@
 @RegisterRowMapper(DatasourceRowMapper.class)
 public interface DatasourceDao {
   @SqlUpdate(
-      "INSERT INTO datasources (guid, name, connection_url) "
-          + "VALUES (:uuid, :name, :connectionUrl)")
-  void insert(@BindBean DatasourceRow dataSourceRow);
+      "INSERT INTO datasources (guid, urn, name, connection_url) "
+          + "VALUES (:uuid, :urn, :name, :connectionUrl)")
+  void insert(@BindBean DatasourceRow datasourceRow);
 
   @SqlQuery("SELECT * FROM datasources WHERE guid = :uuid")
   Optional<DatasourceRow> findBy(@Bind("uuid") UUID uuid);
 
+  @SqlQuery("SELECT * FROM datasources WHERE urn = :urn")
+  Optional<DatasourceRow> findBy(@Bind("urn") String urn);
```

Comment:
Use the `DatasourceUrn` type instead of a string

Reply:
👍